fili-core/src/main/java/com/yahoo/bard/webservice/table/MetricUnionCompositeTable.java

```diff
@@ -55,13 +56,15 @@
      * @param physicalTables  A set of PhysicalTables that are put together under this table. The tables shall have
      * zoned time grains that all satisfy the provided timeGrain
      * @param logicalToPhysicalColumnNames  Mappings from logical to physical names
+     * @param availabilitiesToMetricNames  A map of all availabilities to set of metric names
      */
     public MetricUnionCompositeTable(
             @NotNull TableName name,
             @NotNull ZonedTimeGrain timeGrain,
             @NotNull Set<Column> columns,
             @NotNull Set<ConfigPhysicalTable> physicalTables,
-            @NotNull Map<String, String> logicalToPhysicalColumnNames
+            @NotNull Map<String, String> logicalToPhysicalColumnNames,
+            @NotNull Map<Availability, Set<String>> availabilitiesToMetricNames
```

Comment:
We should create a new constructor and mark this old one as "deprecated" for backward compatibility.

Reply:
More importantly, this constructor simply doesn't work.  It's better to have a compile-time failure of a missing method than a run time failure later.